webapp/portlet/src/main/webapp/activity-composer-app/extension.js

```diff
@@ -9,6 +10,19 @@ export function getActivityComposerExtensions() {
   return activityComposerActions;
 }
 
+export function getActivityComposerHintActionExtensions() {
+  activityComposerHintListActions = getExtensionsByType('activity-composer-hint-action');
+  if (activityComposerHintListActions.length > 0 && activityComposerHintAction.length === 0) {
+    activityComposerHintAction.push(activityComposerHintListActions[0]);
+  }
+  for (let i = 1; i < activityComposerHintListActions.length; i++) {
+    if (activityComposerHintAction.rank > activityComposerHintListActions[i].rank) {
+      activityComposerHintAction = activityComposerHintListActions[i];
+    }
+  }
```

Comment:
Can you explain the logic please ?

Reply:
To ensure that the action take the least rankType a message

Reply:
To ensure that the action take the least rank

Reply:
I see you simplified the logic, but it does not take the rank in account anymore, right ?